### PR TITLE
[serve] Restore "[serve] Support working_dir in serve run (#22760)"

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

#22956 reverted #22760 due to a compatibility issue. This change fixes the issue and restores #22760.

## Related issue number

#22956, #22760

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - This change adds `test_get_job_submission_client_cluster_info` test to ensure the issue doesn't reoccur.

### CODE REVIEW

File: python/ray/serve/scripts.py [hunk 1]
```diff
@@ -173,7 +186,27 @@ def start(http_host, http_port, http_location, checkpoint_path):
...
+    type=str,
+    help='Ray namespace to connect to. Defaults to "serve".',
+)
+def shutdown(address: str, namespace: str):
+    ray.init(
+        address=address,
+        namespace=namespace,
+    )
     serve.api._connect()
     serve.shutdown()
```
Comment (edoakes) [inline python/ray/serve/scripts.py:211]: this should probably go through the REST API right?

File: python/ray/serve/scripts.py [hunk 2]
```diff
@@ -188,14 +221,48 @@ class may or may not be decorated with ``@serve.deployment``.
     hidden=True,
 )
 @click.argument("deployment")
+@click.option(
+    "--address",
+    "-a",
```
Comment (edoakes) [inline python/ray/serve/scripts.py:226]: should we delete these CLI command entirely now?

File: python/ray/serve/scripts.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/scripts.py:211]: actually, we have delete, we should get rid of this one

File: python/ray/serve/scripts.py [hunk 3]
```diff
@@ -220,52 +293,117 @@ def create_deployment(deployment: str, options_json: str):
...
+            del runtime_env_updates["working_dir"]
+
+        # Create ray.init()'s runtime_env
+        if working_dir is None:
+            ray_runtime_env = {}
+        else:
+            ray_runtime_env = {"working_dir": working_dir}
+            if "working_dir" in runtime_env_updates:
+                del runtime_env_updates["working_dir"]
+
```
Comment (edoakes) [inline python/ray/serve/scripts.py:406]: this logic is a bit hard to follow, could you pull it out into a utility and add some unit tests for it?

**Review by edoakes — COMMENTED** [review]
Looks good, just a few small comments. About removing the old commands, don't worry about it in this PR just more general comments.

File: python/ray/serve/scripts.py (hunk 3 above)
Comment (shrekris-anyscale) [reply python/ray/serve/scripts.py]: That logic was needlessly complicated, so I simplified it. Let me know if it's still hard to follow.

**Review by simon-mo — APPROVED** [review]
